### DISCUSSION TITLE
ALTER TABLE (or equivalent)

### DISCUSSION THREAD

**Discussion by FactEngineCommunity** [kuzudb/kuzu#1715] [Q&A] [answered]
Hi there. Is there a feature in Kuzu to **ALTER TABLE** to **add a Property**?

The reason I ask is because:

1. We have:

               CREATE NODE TABLE User(name STRING, age INT64, PRIMARY KEY (name))

2. The Relational Knowledge Graph paradigm is becoming prevalent (e.g. with databases such as EdgeDB, Relational.AI, Apache AGE, AgensGraph) and it is no big thing these days to consider a graph database as if it were relational (e.g. You can hook Neo4j up to JDBC, Relational.AI is implementing DuckDB for ODBC connections etc). I.e. NoSQL, doesn't necessarily mean, to my mind, that we don't have to have a schema, more that schema is optional.

         https://relational.ai/blog/relational-knowledge-graphs
         https://towardsdatascience.com/what-is-a-relational-knowledge-graph-bb747b27ff3f

3. It seems to fit with the feature/request of having PRAGMAs to describe the schema (listing the tables etc). A PRAGMA could describe the Properties on a Table. https://github.com/kuzudb/kuzu/issues/1708

4. Even Neo4j are starting to talk schema: https://neo4j.com/developer-blog/describing-property-graph-data-model/

5. I have a business case for wanting it. Our Boston conceptual modelling software lets you draw a Property Graph Schema, with and as an actual schema.

Please forgive me if the feature already exists. I perceive that one could just add a row with an extra Property to dynamically add a Property to a Node Table...but this request is to explicitly add a Property to a Node Table using some sort of "ALTER TABLE" KuzuQL command.

Your thoughts most welcome.

Many thanks,
Victor

**Comment by ray6080** [comment, marked as answer]
Hi Victor, please check out our documentation [here](https://docs.kuzudb.com/cypher/data-definition/alter/). Is this the expected feature that you're asking?

**Reply by FactEngineCommunity** [reply]
Awesome. Thank you. That's exactly what I'm after. Great, thank you.

**Reply by ray6080** [reply]
You're welcome. Please note that we temporarily disabled setting default value when adding a new property due to some storage changes. But we can add this back if the feature requests this.
